src/ui/Code.ts

```diff
@@ -59,124 +59,96 @@ function DECODE_INSTRUCTION(encodedInstruction: string): [any[]] {
 }
 
 function TO_SIGNED_INTEGER(encodedInstruction: string): [any[]] {
-  return [["", "", toSignedInteger(BigInt(encodedInstruction)).toString(10)]];
+  return [["", "", toUnsignedInteger(BigInt(encodedInstruction)).toString(10)]];
```

Comment:
but main function is called `TO_SIGNED...` ?

Reply:
indeed, the following commit fixes that